src/Controls/src/Xaml/MarkupExtensions/OnIdiomExtension.cs

```diff
@@ -13,6 +14,7 @@ namespace Microsoft.Maui.Controls.Xaml
 		 typeof(IValueConverterProvider),
 		 typeof(IXmlLineInfoProvider),
 		 typeof(IConverterOptions)])]
+	[RequiresUnreferencedCode("The OnIdiomExtension is not trim safe. Use OnIdiom<T> instead.")]
```

Comment:
@jonathanpeppers this is the case I'm most worried about of these extensions. The `{OnIdiom ...}` cannot be simplified at compile time and I don't think we can really make it trimming-friendly because of `Activator.CreateInstance(propertyType)` which we call for value types.

Reply:
I wonder if XamlC could transform `OnIdiomExtension` into `OnIdiom<T>` when it can infer `T` from the values passed to it 🤔 It might be worth trying.

Reply:
Issue: https://github.com/dotnet/maui/issues/22142